packages/op-coreutils/src/op_coreutils/bigquery/write.py

```diff
@@ -287,7 +288,7 @@ def upsert_unpartitioned_table(
     Raises:
         ValueError: If the DataFrame is empty or if unique_keys are not in the DataFrame.
     """
-    if "dt" in df.columns:
+    if "dt" in df.columns and not allow_unpartitioned_dt:
```

Comment:
I think we should stick to this convention in our data warehouse. If you see a table with a `dt` column it should imply that it is partitioned by `dt`.  

When unpartitioned data needs to have a date column we could name the column `date`.